apps/examples/testcase/le_tc/kernel/tc_mqueue.c

```diff
@@ -531,19 +677,40 @@ static void tc_mqueue_mq_notify(void)
 	sa.sa_flags = 0;
 	sigaction(SIGUSR1, &sa, NULL);
 
-	TC_ASSERT_EQ_ERROR_CLEANUP("mq_notify", mq_notify(mqdes, &notification), OK, get_errno(), {
-		goto cleanup;
-	}
-						);
-
-	TC_ASSERT_EQ_CLEANUP("mq_send", mq_send(mqdes, s_msg_ptr, 15, prio), OK, {
-		goto cleanup;
-	}
-						 );
+	TC_ASSERT_EQ_ERROR_CLEANUP("mq_notify", mq_notify(mqdes, &notification), OK, get_errno(), goto cleanup);
+	TC_ASSERT_EQ_CLEANUP("mq_send", mq_send(mqdes, s_msg_ptr, 15, prio), OK, goto cleanup);
 
 	sleep(1);
 	TC_ASSERT_EQ("mq_notify", enter_notify_handler, true);
 
+	mq_close(mqdes);
+
+	mqdes = mq_open("noti_queue", O_CREAT | O_RDWR, 0666, 0);
+	TC_ASSERT_NEQ("mq_open", mqdes, (mqd_t)-1);
+
+	notification.sigev_notify = SIGEV_SIGNAL;
+	notification.sigev_signo = SIGUSR1;
+	sa.sa_handler = mq_notify_handler;
+	sa.sa_flags = 0;
+	sigaction(SIGUSR1, &sa, NULL);
+
+	/* Null Message queue descriptor check */
+	TC_ASSERT_EQ_ERROR_CLEANUP("mq_notify", mq_notify(NULL, &notification), ERROR, get_errno(), goto cleanup);
+
+	/* Invalid signal number check */
+	notification.sigev_signo = 33;
+	TC_ASSERT_EQ_ERROR_CLEANUP("mq_notify", mq_notify(mqdes, &notification), ERROR, get_errno(), goto cleanup);
+
+	/* Valid signal number check */
+	notification.sigev_signo = SIGUSR1;
+	TC_ASSERT_EQ_ERROR_CLEANUP("mq_notify", mq_notify(mqdes, &notification), OK, get_errno(), goto cleanup);
+	TC_ASSERT_EQ_ERROR_CLEANUP("mq_notify", mq_notify(mqdes, &notification), ERROR, get_errno(), goto cleanup);
+
+	/* Detaching the notification */
+	TC_ASSERT_EQ_ERROR_CLEANUP("mq_notify", mq_notify(mqdes, NULL), OK, get_errno(), goto cleanup);
+
+	mq_close(mqdes);
+
```

Comment:
Please call mq_unlink at end of test.

Reply:
This causes TC to assert

Reply:
You mean when you add mq_unlink("noti_queue"); after mq_close, you meet crash. Right? Could you explain more details?

Reply:
It asserts with EBUSY, as we have a reference on the inode